engine/classes/Elgg/EmailService.php

```diff
@@ -166,36 +195,259 @@ public function transport(Email $email) {
 		return true;
 	}
 	
+	/**
+	 * Prepare the subject string
+	 *
+	 * @param string $subject initial subject string
+	 *
+	 * @return string
+	 */
+	protected function prepareSubject(string $subject): string {
+		$subject = elgg_strip_tags($subject);
+		$subject = html_entity_decode($subject, ENT_QUOTES, 'UTF-8');
+		// Sanitise subject by stripping line endings
+		$subject = preg_replace("/(\r\n|\r|\n)/", " ", $subject);
+		return trim($subject);
+	}
+	
 	/**
 	 * Build the body part of the e-mail message
 	 *
-	 * @param Email $email Email
+	 * @param MailMessage $message Current message
+	 * @param Email       $email   Email
 	 *
-	 * @return \Laminas\Mime\Message
+	 * @return \Laminas\Mail\Message
 	 */
-	protected function buildMessageBody(Email $email) {
+	protected function setMessageBody(MailMessage $message, Email $email): MailMessage {
```

Comment:
this is not a correct function name, it doesn't set anything. It just prepares stuff

Reply:
disagree... it sets the body on the message on line 259